Unset environment variable for OpenM++ in AAW

To have multiple versions of OpenM++ on our virtual machines in AAW, we need to be able to set the OM_ROOT environment variable. At the global level it's set to /opt/openmpp and that's something namespace users can't change. We would need to unset this so that things would work the way the OpenM++ developers had intended when writing some scripts to open the UI. @goatsweater or @YannCoderre , can either of you help with this?

I haven't been able to make any progress on this. The repo assumes a hard dependency on `docker`, but I've only got `nerdctl` installed. Tried to alias docker to nerctl since they provide pretty much the same command set, but aliases aren't expanded in Makefiles:

```
aaw-kubeflow-containers git:(master) which docker
docker: aliased to nerdctl

aaw-kubeflow-containers git:(master) make build/remote-desktop
# End repo with exactly one trailing slash, unless it is empty
REPO=$(echo "k8scc01covidacr.azurecr.io" | sed 's:/*$:/:' | sed 's:^\s*/*\s*$::') &&\
	IMAGE_NAME="${REPO}remote-desktop:master" && \
	docker build  --rm --force-rm -t $IMAGE_NAME ./output/remote-desktop && \
	echo -n "Built image $IMAGE_NAME of size: " && \
	docker images $IMAGE_NAME --format "{{.Size}}" && \
	echo "::set-output name=full_image_name::$IMAGE_NAME" && \
	echo "::set-output name=image_tag::master" && \
	echo "::set-output name=image_repo::${REPO}"
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
make: *** [build/remote-desktop] Error 1
```

--- a/docker-bits/6_remote-desktop.Dockerfile
+++ b/docker-bits/6_remote-desktop.Dockerfile
@@ -272,24 +272,24 @@ RUN /bin/bash $RESOURCES_PATH/pspp.sh \
 COPY minio-icon.png $RESOURCES_PATH/minio-icon.png
 COPY remote-desktop/minio-launch.py /usr/bin/minio-launch.py
 
-# OpenM++ Install
-ENV OMPP_VERSION="1.9.8"
-ENV OMPP_PKG_DATE="20220323"
-ARG SHA256ompp=9882798fe2738729cac14d8a62cac8c285fca44e12b8b575ec0d0e6b03ab7a02
+# Install OpenM++
+ENV OMPP_VERSION="1.9.9"
+# IMPORTANT: Don't forget to update the version number in the openmpp.desktop file!!
+ENV OMPP_PKG_DATE="20220505"
+ARG SHA256ompp=479a9a79356a4dd331bcc6cf00110d40feecd6c37f004156f9b4739db2e8ae90
 # OpenM++ environment settings
 ENV OMPP_USER=$NB_USER
 ENV OMPP_GROUP=100
 ENV OMPP_UID=$NB_UID
 ENV OMPP_GID=$NB_GID
-# Where OpenM++ should look for files when building models
-ENV OM_ROOT=/opt/openmpp
 # OpenM++ expects sqlite to be installed (not just libsqlite)
-RUN apt-get install --yes sqlite3
-RUN wget https://github.com/openmpp/main/releases/download/v${OMPP_VERSION}/openmpp_ubuntu_${OMPP_PKG_DATE}.tar.gz -O /tmp/ompp.tar.gz \
+RUN apt-get install --yes sqlite3 \
+    && wget https://github.com/openmpp/main/releases/download/v${OMPP_VERSION}/openmpp_ubuntu_${OMPP_PKG_DATE}.tar.gz -O /tmp/ompp.tar.gz \
     && echo "${SHA256ompp} /tmp/ompp.tar.gz" | sha256sum -c - \
     && tar -xf /tmp/ompp.tar.gz -C /tmp/ \
-    && mv /tmp/openmpp_ubuntu_${OMPP_PKG_DATE} $OM_ROOT \
-    && chown -R $NB_UID:$NB_GID $OM_ROOT
+    && mkdir /opt/openmpp \
+    && mv /tmp/openmpp_ubuntu_${OMPP_PKG_DATE} /opt/openmpp/${OMPP_VERSION} \
+    && chown -R $NB_UID:$NB_GID /opt/openmpp
 # Copy the desktop icon into place for the web UI
 COPY openmpp.png $RESOURCES_PATH/openmpp.png
 

--- a/output/remote-desktop/Dockerfile
+++ b/output/remote-desktop/Dockerfile
@@ -414,24 +414,24 @@ RUN /bin/bash $RESOURCES_PATH/pspp.sh \
 COPY minio-icon.png $RESOURCES_PATH/minio-icon.png
 COPY remote-desktop/minio-launch.py /usr/bin/minio-launch.py
 
-# OpenM++ Install
-ENV OMPP_VERSION="1.9.8"
-ENV OMPP_PKG_DATE="20220323"
-ARG SHA256ompp=9882798fe2738729cac14d8a62cac8c285fca44e12b8b575ec0d0e6b03ab7a02
+# Install OpenM++
+ENV OMPP_VERSION="1.9.9"
+# IMPORTANT: Don't forget to update the version number in the openmpp.desktop file!!
+ENV OMPP_PKG_DATE="20220505"
+ARG SHA256ompp=479a9a79356a4dd331bcc6cf00110d40feecd6c37f004156f9b4739db2e8ae90
 # OpenM++ environment settings
 ENV OMPP_USER=$NB_USER
 ENV OMPP_GROUP=100
 ENV OMPP_UID=$NB_UID
 ENV OMPP_GID=$NB_GID
-# Where OpenM++ should look for files when building models
-ENV OM_ROOT=/opt/openmpp
 # OpenM++ expects sqlite to be installed (not just libsqlite)
-RUN apt-get install --yes sqlite3
-RUN wget https://github.com/openmpp/main/releases/download/v${OMPP_VERSION}/openmpp_ubuntu_${OMPP_PKG_DATE}.tar.gz -O /tmp/ompp.tar.gz \
+RUN apt-get install --yes sqlite3 \
+    && wget https://github.com/openmpp/main/releases/download/v${OMPP_VERSION}/openmpp_ubuntu_${OMPP_PKG_DATE}.tar.gz -O /tmp/ompp.tar.gz \
     && echo "${SHA256ompp} /tmp/ompp.tar.gz" | sha256sum -c - \
     && tar -xf /tmp/ompp.tar.gz -C /tmp/ \
-    && mv /tmp/openmpp_ubuntu_${OMPP_PKG_DATE} $OM_ROOT \
-    && chown -R $NB_UID:$NB_GID $OM_ROOT
+    && mkdir /opt/openmpp \
+    && mv /tmp/openmpp_ubuntu_${OMPP_PKG_DATE} /opt/openmpp/${OMPP_VERSION} \
+    && chown -R $NB_UID:$NB_GID /opt/openmpp
 # Copy the desktop icon into place for the web UI
 COPY openmpp.png $RESOURCES_PATH/openmpp.png
 

--- a/output/remote-desktop/desktop-files/openmpp.desktop
+++ b/output/remote-desktop/desktop-files/openmpp.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=OpenM++
-Exec=/opt/openmpp/bin/ompp_ui_linux.sh
+Exec=/opt/openmpp/1.9.9/bin/ompp_ui_linux.sh
 Icon=/resources/openmpp.png
 Terminal=false
 Categories=Development;

--- a/resources/remote-desktop/desktop-files/openmpp.desktop
+++ b/resources/remote-desktop/desktop-files/openmpp.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=OpenM++
-Exec=/opt/openmpp/bin/ompp_ui_linux.sh
+Exec=/opt/openmpp/1.9.9/bin/ompp_ui_linux.sh
 Icon=/resources/openmpp.png
 Terminal=false
 Categories=Development;

